src/foam/box/SessionServerBox.java

```diff
@@ -42,10 +49,15 @@ public void send(Message msg) {
         session.setLastUsed(new Date());
         session.setUses(session.getUses()+1);
 
-        dao.put(session);
+        sessionDAO.put(session);
 
         if ( authenticate_ && session.getUserId() == 0 ) {
-          msg.replyWithException(new java.security.AccessControlException("not logged in"));
+          msg.replyWithException(new AccessControlException("Not logged in"));
+          return;
+        }
+
+        if ( ! auth.check(getX(), "service." + spec.getName()) ) {
+          msg.replyWithException(new NoPermissionException("Invalid permissions"));
```

Comment:
Is it really an "invalid" permission or just "no" permission or a "missing" permission?

Reply:
It's no permission. Changed it.